unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -206,6 +207,198 @@ public UTF8String toLowerCase() {
     return fromString(toString().toLowerCase());
   }
 
+  /**
+   * Copy the bytes from the current UTF8String, and make a new UTF8String.
+   * @param start the start position of the current UTF8String in bytes.
+   * @param end the end position of the current UTF8String in bytes.
+   * @return a new UTF8String in the position of [start, end] of current UTF8String bytes.
+   */
+  private UTF8String copyUTF8String(byte[] bytes, int start, int end) {
+    int len = end - start + 1;
+    byte[] newBytes = new byte[len];
+    System.arraycopy(bytes, start, newBytes, 0, len);
+    return UTF8String.fromBytes(newBytes);
+  }
+
+  public UTF8String trim() {
+    byte[] bytes = getBytes();
```

Comment:
You can access it each byte by `getByte()`, and copy part of it by `copyMemory()`, so you don't need getBytes() for all these functions.